include/os/freebsd/spl/sys/ccompile.h

```diff
@@ -179,6 +181,9 @@ typedef	void zfs_kernel_param_t;
 #define	MUTEX_NOLOCKDEP 0
 #define	RW_NOLOCKDEP 0
 
+#if !defined(LOCORE) && !defined(_ASM)
```

Comment:
Do I understand correctly that the `#if !defined(LOCORE) && !defined(_ASM)` check is used to detect if this code is not part of a vendor tree?  Assuming that's the case how does this differ from `BUILDING_ZFS` which is defined in the `Makefile.bsd`?  A comment for both of these cases would be helpful.

Reply:
It's used because it's getting pulled in to assembly compilation